File: 2724-convert-an-array-into-a-2d-array-with-conditions/2724-convert-an-array-into-a-2d-array-with-conditions.rb
# @param {Integer[]} nums
# @return {Integer[][]}
def find_matrix(nums)
    map = Array.new(201, 0)
    res = []

    for n in nums
        map[n] += 1
        if map[n] > res.length()
            res.append([n])
        else
            res[map[n] - 1].append(n)
        end
    end
    
    return res
end

File: 2724-convert-an-array-into-a-2d-array-with-conditions/2724-convert-an-array-into-a-2d-array-with-conditions.rs
impl Solution {
    pub fn find_matrix(nums: Vec<i32>) -> Vec<Vec<i32>> {
        let mut map = [0; 201];
        let mut res = vec![];
        for n in nums {
            let n = n as usize;
            map[n] += 1;
            if map[n] > res.len() {
                res.push(vec![n as i32]);
            } else {
                res[map[n] - 1].push(n as i32);
            }
        }

        res
    }
}